docker.go

```diff
@@ -232,8 +280,42 @@ func (p Plugin) Exec() error {
 		cmd.Stdout = os.Stdout
 		cmd.Stderr = os.Stderr
 		trace(cmd)
+		var err error
+		if isCommandBuildxBuild(cmd.Args) && p.CacheMetricsFile != "" {
+			// Create a tee writer and get the channel
+			teeWriter, statusCh := Tee(os.Stdout)
+
+			var goroutineErr error
+
+			var wg sync.WaitGroup
+			wg.Add(1)
+			// Run the command in a goroutine
+			go func() {
+				defer teeWriter.Close()
+				defer wg.Done()
+
+				cmd.Stdout = teeWriter
+				cmd.Stderr = teeWriter
+				goroutineErr = cmd.Run()
+			}()
+
+			// Run the parseCacheMetrics function and handle errors
+			cacheMetrics, err := parseCacheMetrics(statusCh)
+			if err != nil {
+				fmt.Printf("Could not parse cache metrics: %s", err)
+			}
+			wg.Wait()
 
-		err := cmd.Run()
+			if goroutineErr != nil {
+				return goroutineErr
+			}
+
+			if err := writeCacheMetrics(cacheMetrics, p.CacheMetricsFile); err != nil {
```

Comment:
You should only write when you're able to pass successfully

Reply:
Done